ProcessLib/DeactivatedSubdomain.cpp

```diff
@@ -0,0 +1,170 @@
+/**
+ *
+ * \copyright
+ * Copyright (c) 2012-2018, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * File:   DeactivatedSubdomain.cpp
+ *
+ * Created on November 29, 2018, 10:50 AM
+ */
+#include "DeactivatedSubdomain.h"
+
+#include <logog/include/logog.hpp>
+
+#include "BaseLib/ConfigTree.h"
+#include "BaseLib/Error.h"
+#include "BaseLib/TimeInterval.h"
+
+#include "MeshLib/MeshEditing/DuplicateMeshComponents.h"
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Elements/Element.h"
+#include "MeshLib/Node.h"
+
+namespace ProcessLib
+{
+const std::string DeactivatedSubdomain::name_of_paramater_of_zero =
+    "zero_for_element_deactivation_approach";
+
+DeactivetedSubdomainMesh::DeactivetedSubdomainMesh(
+    std::unique_ptr<MeshLib::Mesh> deactivated_subdomain_mesh_,
+    std::vector<MeshLib::Node*>&& inactive_nodes_)
+    : mesh(std::move(deactivated_subdomain_mesh_)),
+      inactive_nodes(std::move(inactive_nodes_))
+{
+}
+
+DeactivatedSubdomain::DeactivatedSubdomain(
+    std::unique_ptr<BaseLib::TimeInterval> time_interval_,
+    std::vector<int>&& materialIDs_,
+    std::vector<std::unique_ptr<DeactivetedSubdomainMesh>>&&
+        deactivated_sudomain_meshes_)
+    : time_interval(std::move(time_interval_)),
+      materialIDs(std::move(materialIDs_)),
+      deactivated_sudomain_meshes(std::move(deactivated_sudomain_meshes_))
+{
+}
+
+bool DeactivatedSubdomain::includesTimeOf(double const t) const
+{
+    return time_interval->contains(t);
+}
+
+std::unique_ptr<DeactivatedSubdomain const> createDeactivatedSubdomain(
+    BaseLib::ConfigTree const& config, MeshLib::Mesh const& mesh)
+{
+    //! \ogs_file_param{prj__process_variables__process_variable__deactivated_subdomains__deactivated_subdomain__time_interval}
+    config.peekConfigParameter<std::string>("time_interval");
+    auto time_interval = BaseLib::createTimeInterval(config);
+
+    std::vector<int> deactivated_subdomain_material_ids;
+    deactivated_subdomain_material_ids =
+        //! \ogs_file_param{prj__process_variables__process_variable__deactivated_subdomains__deactivated_subdomain__material_ids}
+        config.getConfigParameter<std::vector<int>>("material_ids",
+                                                    std::vector<int>{});
+    if (!deactivated_subdomain_material_ids.empty())
```

Comment:
Invert the if condition => reduced indentation

Reply:
done.